src/validation.cpp

```diff
@@ -5002,6 +5010,16 @@ bool ChainstateManager::ActivateSnapshot(
     return true;
 }
 
+CChainState* ChainstateManager::GetChainstateForNewBlock(const uint256& blockhash)
+{
+    if (m_snapshot_chainstate &&
+            !m_snapshot_chainstate->m_chain.Contains(LookupBlockIndex(blockhash))) {
+        return m_snapshot_chainstate.get();
```

Comment:
In commit "validation: introduce ChainstateManager::GetChainstateForNewBlock" (6d85241c236f563715d4a982651fd1815d76dbd2)

Like other places, callers never seem to be checking for null. This should just use references and return `*m_snapshot_chainstate` instead of `m_snapshot_chainstate.get()`.